#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_use-mkoctfile.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Build plplot_octave.oct using mkoctfile and not as a shared library

@DPATCH@

--- plplot-5.8.0~RC1.orig/bindings/octave/CMakeLists.txt
+++ plplot-5.8.0~RC1/bindings/octave/CMakeLists.txt
@@ -133,64 +133,34 @@
 ${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.h
 )
 
-# Build octave interface.
-set(octave_interface_INCLUDE_PATHS
-${CMAKE_SOURCE_DIR}/include
-${CMAKE_BINARY_DIR}
-${CMAKE_BINARY_DIR}/include
-${CMAKE_CURRENT_SOURCE_DIR}
-${CMAKE_CURRENT_BINARY_DIR}
-${OCTAVE_INCLUDE_PATH}
-)
-include_directories(${octave_interface_INCLUDE_PATHS})
-
-add_library(plplot_octave MODULE ${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.cc)
-target_link_libraries(
-plplot_octave
-plplot${LIB_TAG}
-"${OCTAVE_LIBRARIES}"
-"${OCTINTERP_LIBRARIES}"
-)
-
-if(USE_RPATH)
-  get_target_property(LIB_INSTALL_RPATH plplot${LIB_TAG} INSTALL_RPATH)
-  # (Reasonable) assumption here is that OCTAVE_LIBRARIES and
-  # OCTINTERP_LIBRARIES have the same path.
-  get_filename_component(OCTAVE_INSTALL_RPATH "${OCTAVE_LIBRARIES}" PATH)
-  set(LIB_INSTALL_RPATH ${LIB_INSTALL_RPATH} ${OCTAVE_INSTALL_RPATH})
-  set_target_properties(
-  plplot_octave
-  PROPERTIES
-  PREFIX "" 
-  SUFFIX ".oct"
-  INSTALL_RPATH "${LIB_INSTALL_RPATH}"
-  INSTALL_NAME_DIR "${OCTAVE_OCT_DIR}"
-  )
-else(USE_RPATH)
-  set_target_properties(
-  plplot_octave
-  PROPERTIES
-  PREFIX "" 
-  SUFFIX ".oct"
-  INSTALL_NAME_DIR "${OCTAVE_OCT_DIR}"
-  )
-endif(USE_RPATH)
-
 # Have to be specific about permissions for some reason (probably oct suffix).
 set(PERM_MODULES
 OWNER_READ
 OWNER_WRITE
 OWNER_EXECUTE
 GROUP_READ
 GROUP_EXECUTE
 WORLD_READ
 WORLD_EXECUTE
 )
 
+add_custom_command(
+OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.oct
+DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.cc
+COMMAND ${MKOCTFILE} ${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.cc
+-I${CMAKE_CURRENT_BINARY_DIR} -I${CMAKE_SOURCE_DIR}/bindings/octave
+-I${CMAKE_SOURCE_DIR}/include --strip
+-L${CMAKE_BINARY_DIR}/src -lplplot${LIB_TAG}
+)
+
+add_custom_target(
+plplot_octave_oct_file ALL
+DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.oct
+)
+ 
 install(
-TARGETS
-plplot_octave
-LIBRARY
+FILES
+${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.oct
 DESTINATION ${OCTAVE_OCT_DIR}
 PERMISSIONS ${PERM_MODULES}
 )
@@ -238,23 +208,4 @@
 DESTINATION ${PLPLOT_OCTAVE_DIR}
 )
 
-if(never)
-add_custom_command(
-OUTPUT 
-${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.oct
-${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.o
-COMMAND
-touch
-${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.oct
-${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.o
-DEPENDS
-${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.cc
-)
-
-# Link octave interface.
-add_custom_target(
-plplot_octave-libdir.oct ALL
-DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/plplot_octave.o
-)
-endif(never)
 endif(ENABLE_octave)
